includes/wikia/models/DesignSystemGlobalNavigationModel.class.php

```diff
@@ -0,0 +1,314 @@
+<?php
+
+class DesignSystemGlobalNavigationModel extends WikiaModel {
+	const DEFAULT_LANG = 'en';
+
+	private $wikiId;
+	private $lang;
+
+	public function __construct( $wikiId, $lang = self::DEFAULT_LANG ) {
+		parent::__construct();
+
+		$this->wikiId = $wikiId;
+		$this->lang = $lang;
+	}
+
+	public function getData() {
+		global $wgUser;
+
+		$data = [
+			'logo' => [
+				'links' => [
+					[
+						'type' => 'link-image',
+						'href' => $this->getHref( 'fandom-logo' ),
+						'image' => 'wds-company-logo-fandom-powered-by-wikia',
+						'title' => [
+							'type' => 'text',
+							'value' => 'Fandom powered by Wikia'
+						]
+					]
+				]
+			],
+			'search' => $this->getSearchData(),
+			'create_wiki' => [
+				'links' => [
+					[
+						'type' => 'link-text',
+						'title' => [
+							'type' => 'translatable-text',
+							'key' => 'global-navigation-create-wiki-link-start-wikia'
+						],
+						'href' => $this->getHref( 'create-new-wiki' ),
+					]
+				]
+			]
+		];
+
+		if ( $this->lang === static::DEFAULT_LANG ) {
+			$data[ 'fandom_overview' ] = $this->getVerticalsSection();
+			$data[ 'wikis' ] = [
+				'header' => [
+					'type' => 'line-text',
+					'title' => [
+						'type' => 'translatable-text',
+						'key' => 'global-navigation-wikis-header',
+					],
+				],
+				'links' => [
+					[
+						'type' => 'link-text',
+						'title' => [
+							'type' => 'translatable-text',
+							'key' => 'global-navigation-wikis-explore'
+						],
+						'href' => $this->getHref( 'fan-communities' ),
+					],
+					$this->getCommunityCentralLink(),
+					[
+						'type' => 'link-text',
+						'title' => [
+							'type' => 'translatable-text',
+							'key' => 'global-navigation-wikis-fandom-university'
+						],
+						'href' => $this->getHref( 'fandom-university' ),
+					]
+				]
+			];
+		} else {
+			$data[ 'wikis' ] = [
+				'links' => $this->getCommunityCentralLink()
+			];
+		}
+
+		if ( $wgUser->isLoggedIn() ) {
+			$data[ 'user' ] = $this->getLoggedInUserData( $wgUser );
+			$data[ 'notifications' ] = $this->getNotifications( $wgUser );
+		} else {
+			$data[ 'anon' ] = $this->getAnonUserData();
+		}
+
+		return $data;
+	}
+
+	private function getHref( $hrefKey ) {
+		return DesignSystemSharedLinks::getInstance()->getHref( $hrefKey, $this->lang );
+	}
+
+	private function getPageUrl( $pageTitle, $namespace, $query = '' ) {
+		return GlobalTitle::newFromText( $pageTitle, $namespace, $this->wikiId )->getFullURL( $query );
+	}
+
+	private function getSearchData() {
+		$isCorporatePage = WikiaPageType::isCorporatePage();
+
+		$search = [
+			'type' => 'search',
+			'results' => [
+				'url' => $isCorporatePage
+					? $this->getCorporatePageSearchUrl()
+					: $this->getPageUrl( 'Search', NS_SPECIAL, [ 'fulltext' => 'Search' ] ),
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "Search" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)